Add lesson14 (#13)
ReDI-School/js-berlin-2021-fall: lesson14.md

- initialization: a starting point
- condition: a test to see if the loop should stop
- update: changes the value checked by the condition
- action: what we want to repeat

---

### Loops: cake recipe

- initialization: no flour in the bowl (`cupsOfFlour = 0`)
- condition: did we put less than 5 cups of flour? (`cupsOfFlour < 5`)
- update: cups of flour in the bowl (`cupsOfFlour++`)
- action: add flour to the bowl

---

### While loop

Let's see how to make an actual loop in JavaScript.

A `while` loop repeats its body while its `condition` is `true`:

```js
while (condition) {
  // body is executed while "condition" is true
  console.log("Hello from while loop");
}
```

---

### While loop

```js
let cupsOfFlour = 0; // initialization
while (cupsOfFlour < 5) { // condition
    console.log("Take a cup of flour and put it in the bowl");
    cupsOfFlour++; // update
    console.log("There are now " + cupsOfFlour + " cups of flour in the bowl");
}
```

---

### While loop

```js
let bitesOfChocolate = 0;

while (bitesOfChocolate < 10) {
  bitesOfChocolate++;
  console.log("Bites of chocolate: " + bitesOfChocolate);
  console.log("Still hungry.");
}
```

> As long as we have had less than 10 bites of chocolate, we will have one more and console.log that we're still hungry.
> 

Reviewer: ```suggestion
As long as we have had less than 10 bites of chocolate, we will have one more bite and we will still be hungry.
```
Maybe also replace "had" with eaten, used or consumed to illustrate that "you are still hungry until you eat all the bites needed to not feel hungry anymore"? Also, do we want to log "Still hungry" after the last bite? I assume that after the last bite, you're no longer hungry. It might be a bit confusing to add a condition for this case though.
